### PR TITLE
tests: benchmarks: multicore: Add test for S2RAM and STM log

### PR DESCRIPTION
**Pull request by nordic-segl**
Add functional check (logs printed after leaving low power state) for sample application that uses STM logging and tries to enter low power state.

Entering low power state can't be achieved on some cores because of STM logging:
- cpuapp as it handles ETR buffer;
- TDD power domain is kept on;
- etc.

Confirm that when STM logging is used:
- cpu may enter low power states like IDLE or S2RAM;
- STM logging works after leaving low power state.

### CODE REVIEW

File: tests/benchmarks/multicore/idle_stm/testcase.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      remote_rad_CONF_FILE=prj_s2ram.conf
+      remote_ppr_CONF_FILE=prj_s2ram.conf
+      idle_stm_CONFIG_TEST_SLEEP_DURATION_MS=500
+      remote_rad_CONFIG_TEST_SLEEP_DURATION_MS=500
+      remote_ppr_CONFIG_TEST_SLEEP_DURATION_MS=500
+    harness: pytest
+    harness_config:
+      fixture: ppk_power_measure
+      pytest_root:
+        - "${CUSTOM_ROOT_TEST_DIR}/test_measure_power_consumption.py::test_measure_and_data_dump_pwm_and_idle"
```
Comment (nordic-piks) [inline tests/benchmarks/multicore/idle_stm/testcase.yaml:38]: pwm?

**Review by nordic-piks — APPROVED** [review]

File: tests/benchmarks/multicore/idle_stm/sysbuild.cmake [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  BOARD ${SB_CONFIG_BOARD}/${SB_CONFIG_SOC}/cpurad
+  BOARD_REVISION ${BOARD_REVISION}
+)
+
+# ExternalZephyrProject_Add(
+#   APPLICATION remote_ppr
+#   SOURCE_DIR  ${APP_DIR}/remote
+#   BOARD ${SB_CONFIG_BOARD}/${SB_CONFIG_SOC}/cpuppr
+#   BOARD_REVISION ${BOARD_REVISION}
+# )
```
Comment (nordicjm) [inline tests/benchmarks/multicore/idle_stm/sysbuild.cmake:20]: can go

File: tests/benchmarks/multicore/idle_stm/testcase.yaml (hunk 1 above)
Comment (nordic-segl) [reply tests/benchmarks/multicore/idle_stm/testcase.yaml]: Current measurements not needed as SOC can't enter deepest low power states due to use of STM.

File: tests/benchmarks/multicore/idle_stm/boards/nrf54h20dk_nrf54h20_cpuapp.overlay [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/*&cpuppr_vpr {
```
[the remark this answers is not in the document]
Comment (nordic-segl) [reply tests/benchmarks/multicore/idle_stm/boards/nrf54h20dk_nrf54h20_cpuapp.overlay]: this was just for testing. It's uncommented now.

File: tests/benchmarks/multicore/idle_stm/sysbuild.cmake (hunk 2 above)
Comment (nordic-segl) [reply tests/benchmarks/multicore/idle_stm/sysbuild.cmake]: this was just for testing. It's uncommented now.

**Comment by nordic-segl** [comment]
Test was reworked because:
"Setting DBGPWRUPREQ = 1 will keep the TDD power domains on, and also the TnD HSFLL, so it's relatively high power consumption"

**Review by nordicjm — APPROVED** [review]

**Pull request merged**